crates/ruff_linter/src/settings/types.rs

```diff
@@ -635,6 +635,78 @@ impl Deref for CompiledPerFileIgnoreList {
     }
 }
 
+// This struct and its `new` implementation are adapted directly from `PerFileIgnore`, minus the
+// `negated` field
```

Comment:
I don't think this comment is useful for future readers, considering that `PerFileIgnore` has no documentation. It only tells users: Here are two versions of this. 

We should instead document whta this struct does (It specifies the target version for a glob pattern). We could also consider extracting a `PerFile<T>` that's shared between `PerFileIgnore` and `PerFileVersion`.

I'd suggest renaming this struct to `PerFileTargetVersion`. It helps clarifying what kind of version it is.